setup.py

```diff
@@ -27,6 +28,20 @@
 
 is_win32 = sys.platform.startswith('win32')
 
+# On Linux if /tmp is mounted with noexec cythonize will fail
+def is_noexec(mount_point):
+    with open('/proc/mounts', 'r') as mounts_file:
+        for line in mounts_file:
+            fields = line.split()
+            if len(fields) >= 5 and fields[1] == mount_point:
+                options = fields[3].split(',')
+                if 'noexec' in options:
+                    return True
+    return False
+
+if platform.system() == 'Linux' and is_noexec('/tmp'):
+    print("Warning: /tmp is mounted with noexec capacity, cythonize may fail")
```

Comment:
That is problematic:
- it is rather platform specific and might still fail on other platforms due to similar reasons
- the temp dir could be set to something else than /tmp

Reply:
Agreed (I just don't know any better way to do it)